### PR TITLE
Migrate docs to a unified Docusuarus site

### PR DESCRIPTION
**Pull request by shadaj**
Migrate docs to a unified Docusuarus site
Contains the same content as the mdBook, but rendered with Docusaurus for better UI / ability to extend with custom React logic (like the playground).

Only contains the book content for now, but will follow up with additional PRs that migrate the playground / publications page.

---
Stack created with [Sapling](https://sapling-scm.com). Best reviewed with [ReviewStack](https://reviewstack.dev/hydro-project/hydroflow/pull/671).
* #672
* __->__ #671

### CODE REVIEW

**Comment by shadaj** [comment]
This preserves the ability to run mdBook tests, but it seems like they have been broken for a while, so I'm dropping docs from the CI (since we can use Cloudflare Pages to build), and we can open a follow-up to make the tests work.

File: hydroflow/src/lib.rs [hunk 1]
```diff
@@ -28,25 +28,25 @@ pub use hydroflow_macro::*;
...
+    booktest!(example_8_chat_server, quickstart/);
 
-    booktest!(surface_syntax);
-    booktest!(surface_embedding);
-    booktest!(surface_flows);
-    booktest!(surface_data);
+    booktest!(index, syntax/);
+    booktest!(surface_embedding, syntax/);
+    booktest!(surface_flows, syntax/);
+    booktest!(surface_data, syntax/);
```
Comment (MingweiSamuel) [inline hydroflow/src/lib.rs:49]: book tests are run here and pass fine?

File: book.toml
Comment (MingweiSamuel) [inline book.toml:1]: whys this still here?

File: docs/docs/in-out_trees.md
Comment (MingweiSamuel) [inline docs/docs/in-out_trees.md:1]: This file is extra, remove?

File: docs/static/.nojekyll
[the remark this answers is not in the document]
Comment (MingweiSamuel) [reply docs/static/.nojekyll:1]: For gh-pages?

File: book.toml
Comment (MingweiSamuel) [inline book.toml:1]: Ok so can just remove this file entirely I think

File: docs/SUMMARY.md
Comment (MingweiSamuel) [inline docs/SUMMARY.md:1]: I think this is also removable, artifact from mdbook?

File: docs/docs/lattices_crate/index.mdx [hunk 2]
```diff
@@ -1,6 +1,9 @@
+import CrateDocs from '../../../lattices/README.md'
+
+
 # The `lattices` Crate
 
-{{#include ../lattices/README.md:book_include}}
```
Comment (MingweiSamuel) [inline docs/docs/lattices_crate/index.mdx:3]: image

This part is not working quite right

Rustdoc lets you link to rust items without specifying the href https://docs.rs/lattices/latest/lattices/, but doesn't work in normal markdown (https://crates.io/crates/lattices and https://github.com/hydro-project/hydroflow/tree/main/lattices#lattices-1) ....

File: docs/docusaurus.config.js [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+  // metadata like html lang. For example, if your site is Chinese, you may want
+  // to replace "en" with "zh-Hans".
+  i18n: {
+    defaultLocale: 'en',
+    locales: ['en'],
+  },
+
+  stylesheets: [
+    {
+      href: 'https://cdn.jsdelivr.net/npm/katex@0.13.24/dist/katex.min.css',
```
Comment (MingweiSamuel) [inline docs/docusaurus.config.js:48]: would prefer to include via `node_modules` instead but doesnt really matter 🤷‍♀️

File: docs/src/pages/index.js [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+                color: "black",
+                textAlign: "center",
+                background: "white",
+                boxShadow: "0px 3px 8px rgba(0, 0, 0, 0.25)",
+                borderRadius: "15px",
+                width: "200px",
+                margin: "10px",
+                marginTop: 0,
+                fontSize: "1.25em"
+              }}>
```
Comment (MingweiSamuel) [inline docs/src/pages/index.js:52]: Wonder if its worth promoting a link to docs here as well

File: hydroflow/src/lib.rs (hunk 1 above)
Comment (MingweiSamuel) [inline hydroflow/src/lib.rs:49]: nit: wanna swap the args around to make it easier to read
```rust
    macro_rules! booktest {
        ($path:literal, $i:ident) => {
            #[doc = include_str!(concat!("../../docs/docs/", $path, stringify!($i), ".md"))]
            mod $i {}
        };
    }

    booktest!("quickstart/", example_1_simplest);
    booktest!("quickstart/", example_2_simple);
    booktest!("quickstart/", example_3_stream);
    booktest!("quickstart/", example_4_neighbors);
    booktest!("quickstart/", example_5_reachability);
    booktest!("quickstart/", example_6_unreachability);
    booktest!("quickstart/", example_7_echo_server);
    booktest!("quickstart/", example_8_chat_server);

    booktest!("syntax/", index);
    booktest!("syntax/", surface_embedding);
    booktest!("syntax/", surface_flows);
    booktest!("syntax/", surface_data);
```

File: lattices/README.md [hunk 5]
```diff
@@ -1,6 +1,4 @@
-# `lattices`
-
-<!-- ANCHOR: book_include -->
+<h1 class="crate-title"><code>lattices</code></h1>
```
Comment (MingweiSamuel) [inline lattices/README.md:1]: Does ```# `lattices` ``` not work? :P

**Review by MingweiSamuel — APPROVED** [review]
Nice

**Comment by MingweiSamuel** [comment]
Also why does this pr say it has 24,320 additions that seems wrong?

File: docs/docs/lattices_crate/index.mdx (hunk 2 above)
Comment (shadaj) [reply docs/docs/lattices_crate/index.mdx:3]: I will fix this later...

**Pull request merged**